csds-client/client/client_test.go

```diff
@@ -0,0 +1,165 @@
+package client
+
+import (
```

Comment:
I think the tests are not missing some cases. We can add those in subsequent PRs since this one is pretty big in size already.

Reply:
ACK.